### PR TITLE
DOC: update 1.5.0 release notes

### PR DESCRIPTION
**Pull request by tylerjereddy**
* copy over the release notes from the wiki and reformat/clean up
for `rst` format

* use `tools/authors.py` to draft in the author list for release
`1.5.0`

TODO:

- [x] fill in issue/PR lists when `gh_lists.py` is working: #12220
- [x] manually go through `1.5.0` PRs with `ENH` label and fill in missing entries (please do help)
- [x] adjust `.mailmap` for author name issues brought up in this PR [**NO AUTHOR COMPLAINTS AT ALL**]
- [x] finalize the "highlights" section of the release notes

### CODE REVIEW

File: doc/release/1.5.0-notes.rst [hunk 1]
```diff
@@ -39,39 +57,207 @@ New features
...
+analytical formulas instead of numerical integration for greater speed and 
+accuracy
+
+Moments and entropy trapezoidal distribution (`scipy.stats.trapz`) are now 
+calculated using analytical formulas instead of numerical integration for 
+greater speed and accuracy
+
+Methods of the truncated normal distribution (`scipy.stats.truncnorm`), 
+especially ``_rvs``, are significantly faster after a complete rewrite.
+
```
Comment (WarrenWeckesser) [inline doc/release/1.5.0-notes.rst:219]:
```suggestion

The `fit` method of the Laplace distribution,  `scipy.stats.laplace`, now uses the
analytical formulas for the maximum likelihood estimates of the parameters.

```

File: doc/release/1.5.0-notes.rst [hunk 2]
```diff
@@ -39,39 +57,207 @@ New features
...
+`scipy.spatial.SphericalVoronoi` now supports calculation of region areas in 2D 
+and 3D cases
+
+`scipy.special` improvements
+--------------------------
+The inverse error function and complimentary inverse error function now 
+provide a Cython interface and are now accessible as 
+`scipy.special.cython_special.erfinv` and `scipy.special.cython_special.erfcinv`
+respectively.
```
Comment (person142) [inline doc/release/1.5.0-notes.rst:177]:
```suggestion
The following functions now have Cython interfaces in `cython_special`:

- `special.erfinv`
- `special.erfcinv`
- `special.spherical_jn`
- `special.spherical_yn`
- `special.spherical_in`
- `special.spherical_kn`
```

File: doc/release/1.5.0-notes.rst [hunk 3]
```diff
@@ -80,21 +289,170 @@ Backwards incompatible changes
...
+signal and filter lengths.
 
 `scipy.stats` changes
 ---------------------
 
 
 Other changes
 =============
+Improved support for leveraging 64-bit integer size from linear algebra backends
+in several parts of the SciPy codebase.
```
Comment (ev-br) [inline doc/release/1.5.0-notes.rst:311]: This well might be in the highlights

File: doc/release/1.5.0-notes.rst [hunk 4]
```diff
@@ -39,39 +57,230 @@ New features
...
 `scipy.spatial` improvements
 ----------------------------
+`scipy.spatial.geometric_slerp` has been added to enable geometric 
+spherical linear interpolation on an n-sphere
+
+`scipy.spatial.SphericalVoronoi` now supports calculation of region areas in 2D 
+and 3D cases
+
+New ``cKDTree`` benchmarks are available for building and querying of balanced/
+unbalanced kd-trees.
```
Comment (rgommers) [inline doc/release/1.5.0-notes.rst:179]: The benchmarks are nice, but the more interesting part is that the performance improved a lot for some cases - see gh-11653. That should be mentioned here.

File: doc/release/1.5.0-notes.rst [hunk 5]
```diff
@@ -39,39 +57,230 @@ New features
...
 `scipy.stats` improvements
 --------------------------
+The function for generating random samples in `scipy.stats.dlaplace` has been 
+improved. The new function generates random samples by calculating the 
+difference between two independent and identically distributed geometric 
+distributions. This new function is approximately twice as fast as the previous 
+(default) function which generated samples by inverting the CDF. This new 
+function also uses less memory than the previous function. For one sample, the 
+new function uses ~25% less memory; for one million samples, it uses ~40% less 
+memory; for ten million samples, it uses ~60% less memory.
```
Comment (rgommers) [inline doc/release/1.5.0-notes.rst:205]: I'd suggest to shorten this a little. Something like ".... uses between 25% and 60% less memory (see `gh-xxxxx`)"

File: doc/release/1.5.0-notes.rst [hunk 6]
```diff
@@ -39,39 +57,230 @@ New features
...
+The new function ``median_abs_deviation`` replaces the deprecated 
+``median_absolute_deviation``.
+
+The ``wilcoxon`` function now computes the p-value for Wilcoxon's signed rank 
+test using the exact distribution for inputs up to length 25.  The function has 
+a new ``mode`` parameter to specify how the p-value is to be computed.  The 
+default is ``"auto"``, which uses the exact distribution for inputs up to length 
+25 and the normal approximation for larger inputs.
+
+Added a new cython-based implementation to evaluate guassian kernel estimates,
```
Comment (rgommers) [inline doc/release/1.5.0-notes.rst:257]: minor: cython -> Cython

File: doc/release/1.5.0-notes.rst [hunk 7]
```diff
@@ -39,39 +57,230 @@ New features
...
 
 `scipy` deprecations
 --------------------
 
+`scipy.special` changes
+-----------------------
+The ``bdtr``, ``bdtrc``, and ``bdtri`` functions are deprecating non-negative 
+non-integral ``n`` arguments. With this release ``n>0`` is still supported for 
+arbitrary ``n``. However, a deprecation warning is emitted notifying that 
+non-integral n will not be supported with SciPy ``1.7.0`` release.
```
Comment (rgommers) [inline doc/release/1.5.0-notes.rst:274]: I would shorten this to only the first sentence of the paragraph, the rest is fairly redundant.

File: doc/release/1.5.0-notes.rst [hunk 8]
```diff
@@ -80,21 +289,170 @@ Backwards incompatible changes
 
 `scipy.linalg` changes
 ----------------------
+The output signatures of ``?syevr``, ``?heevr`` have been changed from 
+``w, v, info`` to ``w, v, m, isuppz, info``
+
+The order of output arguments ``w``, ``v`` of ``<sy/he>{gv, gvd, gvx}`` is 
+swapped.
```
Comment (rgommers) [inline doc/release/1.5.0-notes.rst:296]: Noting here for further review: check if these give hard errors; sounds like silently changing numerical values now.

File: doc/release/1.5.0-notes.rst [hunk 9]
```diff
@@ -80,21 +289,170 @@ Backwards incompatible changes
...
 =============
+Improved support for leveraging 64-bit integer size from linear algebra backends
+in several parts of the SciPy codebase.
+
+Shims designed to ensure the compatibility of SciPy with Python 2.7 have now 
+been removed.
+
+Many warnings due to unused imports and unused assignments have been addressed.
+
+Early stage adoption of ``mypy`` type hinting in a few parts of the codebase
```
Comment (rgommers) [inline doc/release/1.5.0-notes.rst:318]: It's not specific to mypy. Suggest to change "``mypy`` type hinting" to "type annotations".

**Review by rgommers — COMMENTED** [review]
Thanks Tyler. Looks like another solid release

File: doc/release/1.5.0-notes.rst [hunk 10]
```diff
@@ -80,21 +289,170 @@ Backwards incompatible changes
...
 Other changes
 =============
+Improved support for leveraging 64-bit integer size from linear algebra backends
+in several parts of the SciPy codebase.
+
+Shims designed to ensure the compatibility of SciPy with Python 2.7 have now 
+been removed.
+
+Many warnings due to unused imports and unused assignments have been addressed.
+
```
Comment (rgommers) [inline doc/release/1.5.0-notes.rst:317]: Also worth adding in this section: Many usage examples were added to function docstrings.

File: doc/release/1.5.0-notes.rst [hunk 11]
```diff
@@ -39,39 +57,230 @@ New features
...
+- ``?pteqr``: computes the eigenvectors and eigenvalues of a positive definite 
+tridiagonal matrix
+- ``?tbtrs``: solves a linear system with a triangular banded matrix
+- ``?csd``: computes the Cosine Sine decomposition of an orthogonal/unitary 
+matrix
+
+Generalized QR factorization routines (``?geqrf``) now have full ``_lwork`` 
+counterparts.
+
+`scipy.linalg.cossin` Cosine Sine decomposition of unitary matrices has been
```
[the remark this answers is not in the document]
Comment (ilayn) [reply doc/release/1.5.0-notes.rst:109]: we had this discussion but we couldn't came up with anything better and it is kind of what everybody goes with. So we counted on the fact that it is under linalg namespace. But yes I would really appreciate a better one.

File: doc/release/1.5.0-notes.rst [hunk 12]
```diff
@@ -39,39 +57,230 @@ New features
```
Comment (AtsushiSakai) [inline doc/release/1.5.0-notes.rst:57]: This is for #11736
```suggestion
Input validations and intuitive exception messages are added for ``UnivariateSpline``,  ``RectSphereBivariateSpline``, `RectBivariateSpline``, ``SmoothSphereBivariateSpline``, `` LSQSphereBivariateSpline``, ``SmoothBivariateSpline``, ``LSQBivariateSpline``, ``LSQUnivariateSpline``, ``InterpolatedUnivariateSpline``, and ``UnivariateSpline`` functions.
```

File: doc/release/1.5.0-notes.rst [hunk 13]
```diff
@@ -39,39 +57,230 @@ New features
...
+``differential_evolution`` can now use the modern ``np.random.Generator`` as 
+well as the legacy ``np.random.RandomState`` as a seed.
 
 `scipy.signal` improvements
 ---------------------------
+A new optional argument ``include_nyquist`` is added to ``freqz`` functions in 
+this module. It is used for including the last frequency (Nyquist frequency).
 
+`scipy.signal.find_peaks_cwt` now accepts a ``window_size`` parameter for the 
+size of the window used to calculate the noise floor.
```
Comment (AtsushiSakai) [inline doc/release/1.5.0-notes.rst:158]: This is for #11347
```suggestion
Input validations and intuitive exceptions messages are added for ``buttord``, ``cheb1ord``, ``cheb2ord`` functions.
```

**Review by AtsushiSakai — COMMENTED** [review]
These are the PRs that I created. If there is anything I can do, let me know.

**Comment by tylerjereddy** [comment]
@AtsushiSakai Ok, I condensed the large number of input validation/exception handling improvements to a single broad statement.

File: doc/release/1.5.0-notes.rst [hunk 14]
```diff
@@ -24,12 +24,30 @@ For running on PyPy, PyPy3 6.0+ and NumPy 1.15.0 are required.
 Highlights of this release
 --------------------------
 
+- wrapers for more than a dozen new ``LAPACK`` routines are now available
+  in `scipy.linalg.lapack`
+- `scipy.optimize` ``minimize`` methods using finite difference numerical
```
[the remark this answers is not in the document]
Comment (tylerjereddy) [reply doc/release/1.5.0-notes.rst]: @mdhaber Suggestions for rewording here?

[the remark this answers is not in the document]
Comment (tylerjereddy) [reply doc/release/1.5.0-notes.rst]: I was summarizing 18 lines of release notes content into one highlight--a bunch of stuff got better in `optimize`; if they want to pick the most important of those improvements I'll change it.

**Comment by tylerjereddy** [comment]
Ok, I did the "manual sweep" of `1.5.0` PRs with `enhancement` labels and added a dozen or so additional things that were missed.

Also, tried to fix the doc build errors because of `rst` changes here.

File: doc/release/1.5.0-notes.rst [hunk 15]
```diff
@@ -80,24 +310,813 @@ Backwards incompatible changes
...
+* jeremie du boisberranger +
+* Stefan Endres
+* Malte Esders +
+* Leo Fang +
+* felixhekhorn +
+* Isuru Fernando
+* Andrew Fowlie
+* Lakshay Garg +
+* Gaurav Gijare +
+* GitHub
```
Comment (tylerjereddy) [inline doc/release/1.5.0-notes.rst:383]: user `GitHub` shows up on every release &  I end up deleting manually; should probably figure out if that's leaking from incomplete `.mailmap` or something else

File: doc/release/1.5.0-notes.rst [hunk 16]
```diff
@@ -24,12 +24,35 @@ For running on PyPy, PyPy3 6.0+ and NumPy 1.15.0 are required.
 Highlights of this release
 --------------------------
 
--
+- wrappers for more than a dozen new ``LAPACK`` routines are now available
+  in `scipy.linalg.lapack`
```
Comment (mdhaber) [inline doc/release/1.5.0-notes.rst:28]: Do we want to acknowledge NumFOCUS' support of this effort?  (e.g. "thanks to the support of a NumFOCUS small development grant")

File: doc/release/1.5.0-notes.rst [hunk 17]
```diff
@@ -24,12 +24,35 @@ For running on PyPy, PyPy3 6.0+ and NumPy 1.15.0 are required.
...
+- `scipy.optimize.minimize` methods using finite difference numerical 
+  differentiation have gained several new features and there have been
```
Comment (mdhaber) [inline doc/release/1.5.0-notes.rst:30]:
```suggestion
  differentiation have gained several new features, and there have been
```

File: doc/release/1.5.0-notes.rst [hunk 18]
```diff
@@ -24,12 +24,35 @@ For running on PyPy, PyPy3 6.0+ and NumPy 1.15.0 are required.
...
+  performance improvements in `scipy.optimize.linprog` and minimization
```
Comment (mdhaber) [inline doc/release/1.5.0-notes.rst:31]:
```suggestion
  performance improvements in minimization
```
Thanks, but I don't think those performance enhancements deserve a highlight.  (gh-12043 certainly would, but it looks like that's getting bumped.)

File: doc/release/1.5.0-notes.rst [hunk 19]
```diff
@@ -24,12 +24,35 @@ For running on PyPy, PyPy3 6.0+ and NumPy 1.15.0 are required.
...
 
 
 New features
 ============
 
+`scipy.cluster` improvements
+------------------------------
+Initialization of `scipy.cluster.vq.kmeans2` using ``minit="++"`` had a 
+quadratic complexity in the number of samples. It has been improved, resulting 
+in a now much faster, with quasi-linear complexity, initialization.
```
Comment (mdhaber) [inline doc/release/1.5.0-notes.rst:44]:
```suggestion
in a much faster initialization with quasi-linear complexity.
```

File: doc/release/1.5.0-notes.rst [hunk 20]
```diff
@@ -39,39 +62,246 @@ New features
...
+
+- 2-point, 3-point, or complex step finite differences can be used. Previously 
+  only a 2-step finite difference was available.
+- There is now the possibility to use a relative step size, previously only an
+  absolute step size was available.
+- If the ``minimize`` method uses bounds the numerical differentiation strictly 
+  obeys those limits.
+- The numerical differentiation machinery now makes use of a simple cache, 
+  which in some cases can reduce the number of function evaluations.
+
```
Comment (mdhaber) [inline doc/release/1.5.0-notes.rst:141]: - ``minimize``'s ``method= 'powell'` now supports simple bound constraints

File: doc/release/1.5.0-notes.rst (hunk 16 above)
Comment (tylerjereddy) [reply doc/release/1.5.0-notes.rst:28]: I'm not aware of a precedent for this in our release notes for other things like GSOC, etc., so not sure I'm comfortable adding at this stage.

File: doc/release/1.5.0-notes.rst (hunk 14 above)
[the remark this answers is not in the document]
Comment (tylerjereddy) [reply doc/release/1.5.0-notes.rst]: I've cut that highlight out completely based on the feedback

**Comment by tylerjereddy** [comment]
If the CI is green after most recent update I will most likely merge.

**Comment by tylerjereddy** [comment]
We can consider backporting for unresolved matters if we really have to.

File: doc/release/1.5.0-notes.rst [hunk 21]
```diff
@@ -39,39 +60,247 @@ New features
...
+Added a new Cython-based implementation to evaluate guassian kernel estimates,
+which should improve the performance of ``gaussian_kde``
+
+The ``winsorize`` function now has a ``nan_policy`` argument for refined
+handling of ``nan`` input values.
+
+The ``binned_statistic_dd`` function with ``statistic="std"`` performance was
+improved by ~4x.
+
+``scipy.stats.kstest(rvs, cdf,...)`` now handles both one-sample and
```
Comment (pvanmulbregt) [inline doc/release/1.5.0-notes.rst:270]: perhaps put the 'kstest' higher up next to the 'kstwo' item as they both relate to K-S statistics?

File: doc/release/1.5.0-notes.rst [hunk 22]
```diff
@@ -39,39 +60,247 @@ New features
...
+random number generation (e.g. random variates from distributions) can now use 
+the modern ``np.random.Generator`` as well as the legacy 
+``np.random.RandomState`` as a seed.
+
+The ``axis`` parameter was added to `scipy.stats.rankdata`. This allows slices 
+of an array along the given axis to be ranked independently.
+
+The ``axis`` parameter was added to `scipy.stats.f_oneway`, allowing it compute 
+multiple one-way ANOVA tests for data stored in n-dimensional arrays. Some
+performance improvements to ``f_oneway`` as well.
```
Comment (WarrenWeckesser) [inline doc/release/1.5.0-notes.rst:221]: Add the missing word `to`, and make the last phrase a sentence.
```suggestion
The ``axis`` parameter was added to `scipy.stats.f_oneway`, allowing it to
compute  multiple one-way ANOVA tests for data stored in n-dimensional
arrays.  The performance of ``f_oneway`` was also improved for some cases.
```

File: doc/release/1.5.0-notes.rst (hunk 11 above)
[the remark this answers is not in the document]
Comment (rgommers) [reply doc/release/1.5.0-notes.rst:113]: either `csd` (in analogy with `svd`) or a more explicit `cossin_decomp` would be better imho

File: doc/release/1.5.0-notes.rst (hunk 8 above)
Comment (ilayn) [reply doc/release/1.5.0-notes.rst:316]: Only the argument order is changed to match the remaining ones. So I don't know if we can raise errors based on that or how.

Comment (rgommers) [reply doc/release/1.5.0-notes.rst:316]: They do have different shape at least:
```
w : rank-1 array('d') with bounds (n)
v : rank-2 array('d') with bounds (n,n) and a storage
```

Still, it sounds like something to avoid. Was there a reason why it had to be done (e.g. infrastructure for a whole family of functions), or was it simply "let's make it consistent"?

Comment (ilayn) [reply doc/release/1.5.0-notes.rst:316]: Probably they were hardly ever used. Because their lwork argument was not correct. So both fixed and made consistent.

File: doc/release/1.5.0-notes.rst (hunk 11 above)
[the remark this answers is not in the document]
Comment (ilayn) [reply doc/release/1.5.0-notes.rst:113]: `csd` is already overloaded in [`scipy.signal.csd`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.csd.html) and some circles already use it for a shortcut to cumulative power spectral density. So that won't fly

Frankly, `cossin_decomp` is even uglier than `cossin` also becomes a one-off since the others don't have it. 

Why I prefer `cossin` is that there is nothing called `cossin` in any trig libraries. And you can't accidentally fall to that namespace.

File: doc/release/1.5.0-notes.rst (hunk 8 above)
Comment (rgommers) [reply doc/release/1.5.0-notes.rst:316]: Okay that makes sense - I can try to add a sentence so that that bit doesn't read like people should go audit their code because we did something unusual.
